analysis/monkwindwalker/src/castDamage.ts

```diff
@@ -0,0 +1,34 @@
+import SPELLS from 'common/SPELLS';
+import Spell from 'common/SPELLS/Spell';
+
+/**
+ * Map of casted spells (that can appear when listening to casts) and their
+ * associated damage (that can appear when listening to damage)
+ *
+ * CASTED ability on the left, DAMAGE spell on the right.
+ *
+ * This map only makes sense with 1-to-1 connections.
+ */
+const castDamageMap: ReadonlyMap<Spell, Spell[]> = new Map([
+  [SPELLS.FISTS_OF_FURY_CAST, [SPELLS.FISTS_OF_FURY_DAMAGE]],
+  [SPELLS.WHIRLING_DRAGON_PUNCH_TALENT, [SPELLS.WHIRLING_DRAGON_PUNCH_TALENT_TICK]],
+  [SPELLS.SPINNING_CRANE_KICK, [SPELLS.SPINNING_CRANE_KICK_DAMAGE]],
+  [SPELLS.CHI_BURST_TALENT, [SPELLS.CHI_BURST_TALENT_DAMAGE]],
+  [SPELLS.CHI_WAVE_TALENT, [SPELLS.CHI_WAVE_TALENT_DAMAGE]],
+  [SPELLS.RISING_SUN_KICK, [SPELLS.RISING_SUN_KICK_SECOND]],
+  [
+    SPELLS.FAELINE_STOMP_CAST,
+    [SPELLS.FAELINE_STOMP_PULSE_DAMAGE, SPELLS.FAELINE_STOMP_DAMAGE_AND_HEAL],
+  ],
+]);
```

Comment:
since this is internal and you always use the `entries` anyway *and* you already have it written in entries format, should this just be `const entries = [ .... ];`?

Not sure either way. Up to you how you want to do it.

Reply:
Not sure from github interface which element you're referring to..

`castDamageMap` is a `Map` so I can map from object to object.

`entries` is to get an array from the `Map` since the `Iterator` that `Map.entries()` returns does not have `.map()`.

`castToDamage` and `damageToCast` are simple ID->`Spell` dictionaries to easily get `Spell` with all meta data with IDs.

Reply:
so my thinking is that `castDamageMap` is never actually used as a map (it is not exported, and its only use here is in defining the `entries` const).

so why not change it to:


```ts
const entries = [
  [SPELLS.FISTS_OF_FURY_CAST, [SPELLS.FISTS_OF_FURY_DAMAGE]],
  [SPELLS.WHIRLING_DRAGON_PUNCH_TALENT, [SPELLS.WHIRLING_DRAGON_PUNCH_TALENT_TICK]],
  [SPELLS.SPINNING_CRANE_KICK, [SPELLS.SPINNING_CRANE_KICK_DAMAGE]],
  [SPELLS.CHI_BURST_TALENT, [SPELLS.CHI_BURST_TALENT_DAMAGE]],
  [SPELLS.CHI_WAVE_TALENT, [SPELLS.CHI_WAVE_TALENT_DAMAGE]],
  [SPELLS.RISING_SUN_KICK, [SPELLS.RISING_SUN_KICK_SECOND]],
  [
    SPELLS.FAELINE_STOMP_CAST,
    [SPELLS.FAELINE_STOMP_PULSE_DAMAGE, SPELLS.FAELINE_STOMP_DAMAGE_AND_HEAL],
  ],
];
```

and ditch the map entirely?

Then you use `entries` as you currently are for constructing `castToDamage` and `damageToCast`

Reply:
this is really a nitpick, not going to block merging over it :)

Reply:
That's definitely a good point. But I'd maybe argue that a `Map` is a bit of a clearer structure for someone looking at the code, than just an array of arrays and objects? The syntax for creating Maps are a bit awkward, and I really wish I could make the objects in an easier way than going via entries to array... 

I feel like if you look at the code, a `Map`, while somewhat arcane, at least has a clear purpose, while the type `[Spell, Spell[]][]` is kind of nonsense lol.

Originally I created the map and then util methods that took `id` and then `find()` the entries, but that felt like unnecessary looping for each spell.

Reply:
Dropping the Map for just `entries` is a small perf. optimization at dev logic cost, which I don't think is worth it in this case.

Reply:
:+1: